openmls/src/extensions/required_capabilities.rs

```diff
@@ -31,37 +40,49 @@ use super::{Deserialize, ExtensionError, ExtensionType, Serialize};
     TlsSize,
 )]
 pub struct RequiredCapabilitiesExtension {
-    extensions: Vec<ExtensionType>,
-    proposals: Vec<ProposalType>,
+    extension_types: Vec<ExtensionType>,
+    proposal_types: Vec<ProposalType>,
+    credential_types: Vec<CredentialType>,
 }
 
 impl RequiredCapabilitiesExtension {
     /// Creates a new [`RequiredCapabilitiesExtension`] from extension and proposal types.
-    pub fn new(extensions: &[ExtensionType], proposals: &[ProposalType]) -> Self {
+    pub fn new(
+        extension_types: &[ExtensionType],
+        proposal_types: &[ProposalType],
+        credential_types: &[CredentialType],
+    ) -> Self {
         Self {
-            extensions: extensions.into(),
-            proposals: proposals.into(),
+            extension_types: extension_types.into(),
+            proposal_types: proposal_types.into(),
+            credential_types: credential_types.into(),
         }
     }
 
     /// Get a slice with the required extension types.
-    pub(crate) fn extensions(&self) -> &[ExtensionType] {
-        self.extensions.as_slice()
+    pub(crate) fn extension_types(&self) -> &[ExtensionType] {
+        self.extension_types.as_slice()
     }
 
     /// Get a slice with the required proposal types.
-    pub(crate) fn proposals(&self) -> &[ProposalType] {
-        self.proposals.as_slice()
+    pub(crate) fn proposal_types(&self) -> &[ProposalType] {
+        self.proposal_types.as_slice()
+    }
+
+    /// Get a slice with the required credential types.
+    #[allow(unused)]
+    pub(crate) fn credential_types(&self) -> &[CredentialType] {
+        self.credential_types.as_slice()
     }
 
     /// Check if all extension and proposal types are supported.
     pub(crate) fn check_support(&self) -> Result<(), ExtensionError> {
-        for extension in self.extensions() {
+        for extension in self.extension_types() {
             if !extension.is_supported() {
                 return Err(ExtensionError::UnsupportedExtensionType);
             }
         }
-        for proposal in self.proposals() {
+        for proposal in self.proposal_types() {
```

Comment:
We should probably add this for credentials as well. But not in this PR.

Reply:
I filed https://github.com/openmls/openmls/issues/1205.